smolyay/basis.py

```diff
@@ -295,6 +309,11 @@ def domain(self):
         """numpy.ndarray: Domain the sample points come from."""
         return numpy.array([-1, 1])
 
+    @property
+    def integral_over_domain(self):
+        """float: basis function integrated over the domain."""
+        return (1 - (-1)**(self.degree+1))/(self.degree + 1)
```

Comment:
You can use odd / even to do this one too:
```suggestion
        if self.degree % 2 == 1:
            return 0
        else:
            return 2/(self.degree + 1)
```